components/atom/switch/src/index.js

```diff
@@ -0,0 +1,179 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import cx from 'classnames'
+import AtomLabel from '../../label/src/index'
+
+const BASE_CLASS = 'sui-AtomSwitch'
+
+const SIZES = {
+  MEDIUM: 'medium',
+  LARGE: 'large'
+}
+
+const TYPES = {
+  TOGGLE: 'toggle',
+  SELECT: 'select',
+  SINGLE: 'single'
+}
+
+class AtomSwitch extends Component {
+  constructor (props) {
+    super(props)
+
+    const {size, disabled, labelLeft, labelRight, type, name, label, labelOptionalText, onToggle} = props
+    this.onToggle = onToggle
+    this.size = size
+    this.type = type
+    this.state = {
+      toggle: false,
+      disabled: disabled,
+      isFocus: false,
+      labelLeft: labelLeft,
+      labelRight: labelRight,
+      name: name,
+      label: label,
+      labelOptionalText: labelOptionalText
+    }
+
+    window.addEventListener('keydown', this.keyBindings.bind(this), true)
+  }
+
+  keyBindings = (event) => {
+    if (!this.state.isFocus) {
+      return
+    }
+
+    if (event.defaultPrevented) {
+      return
+    }
+
+    switch (event.key) {
+      case 'Enter':
+        this.toggleSwitch()
+        event.preventDefault()
+        break
+      case ' ':
+        this.toggleSwitch()
+        event.preventDefault()
+        break
+    }
+  }
+
+  toggleSwitch = function () {
+    if (this.state.disabled) {
```

Comment:
To improve readability, we usually declare props as varaibles on the top of the scope:
```
const {disabled, onToggle,toggle} = this.state
```

Reply:
make sense, thanks